pkg/ccl/changefeedccl/changefeed.go

```diff
@@ -64,12 +66,26 @@ func createProtectedTimestampRecord(
 	progress.ProtectedTimestampRecord = uuid.MakeV4()
 	log.VEventf(ctx, 2, "creating protected timestamp %v at %v",
 		progress.ProtectedTimestampRecord, resolved)
-	spansToProtect := makeSpansToProtect(codec, targets)
+	deprecatedSpansToProtect := makeSpansToProtect(codec, targets)
```

Comment:
we need both targets and spans?

Reply:
Until we run the migration in https://github.com/cockroachdb/cockroach/pull/74281 we want to continue protecting spans so that jobs started in a mixed version state do not fail. Once the migration is complete, the `Protect` method in the protected ts `Storage` interface will stop writing these passed in spans to the underlying `system.pts_records` table (this will come as a follow up PR). So all jobs started after the migration might pass in a record with the spans field set, but this will not be persisted.

The idea is that GC for 22.1 will continue to respect both spans protected by the old subsystem, and targets protected by the new subsystem since this simplifies the migration in a mixed version state. In 22.2 with some elbow grease, we should be able to stop populating the spans field in the record entirely.